PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/StorageUploadWithoutLengthCheckTest.java

```diff
@@ -0,0 +1,139 @@
+ * 1. @ServiceMethod(returns = ReturnType.SINGLE)
+ *     public void upload(InputStream data) {
+ *         uploadWithResponse(new BlobParallelUploadOptions(data), null, null);
+ *     }
+ *
+ * 2. uploadWithResponse(new BlobParallelUploadOptions(data).setRequestConditions(blobRequestConditions), null, Context.NONE);
+ *
+ * 3. upload(data, false);
+ */
+
+public class StorageUploadWithoutLengthCheckTest {
+
+    // Declare as instance variables
+    @Mock
+    private ProblemsHolder mockHolder;
+    private PsiElementVisitor mockVisitor;
+    private PsiElement mockElement;
+
+    @BeforeEach
+    public void setup() {
+        mockHolder = mock(ProblemsHolder.class);
+        mockElement = mock(PsiElement.class);
+        mockVisitor = createVisitor();
+    }
+
+    /**
+     * assertVisitor(mockVisitor) asserts that it the visitor not null and is an instance of JavaRecursiveElementWalkingVisitor.
+     * verifyRegisterProblem(mockVisitor, METHOD_TO_CHECK, NUMBER_OF_INVOCATIONS) verifies that the registerProblem method is called
+     * when the visitor visits a method call expression with the method name METHOD_TO_CHECK.
+     */
+    @Test
+    public void testStorageUploadWithoutLengthCheck() {
+        assertVisitor(mockVisitor);
+
+        String METHOD_TO_CHECK = "upload";
+        int NUMBER_OF_INVOCATIONS = 1;  // Number of times registerProblem should be called
+        verifyRegisterProblem(mockVisitor, METHOD_TO_CHECK, NUMBER_OF_INVOCATIONS);
+
+    }
+
+    private PsiElementVisitor createVisitor() {
+        boolean isOnTheFly = false;
+
+        // Create an instance of the class under test
+        StorageUploadWithoutLengthCheck check = new StorageUploadWithoutLengthCheck();
+        return check.buildVisitor(mockHolder, isOnTheFly);
+    }
+
+    void assertVisitor(PsiElementVisitor visitor) {
+        assertNotNull(visitor);
+        assertTrue(visitor instanceof JavaRecursiveElementWalkingVisitor);
+    }
+
+    private void verifyRegisterProblem(PsiElementVisitor visitor, String METHOD_TO_CHECK, int NUMBER_OF_INVOCATIONS) {
+
+        // Arrange
+        PsiMethodCallExpression mockExpression = mock(PsiMethodCallExpression.class);
+        PsiExpressionList mockArgList = mock(PsiExpressionList.class);
+
+        PsiExpression mockArgExpression = mock(PsiExpression.class);
+
+        PsiMethodCallExpression mockArgMethodCall = mock(PsiMethodCallExpression.class);
+        PsiReferenceExpression mockMethodExpression = mock(PsiReferenceExpression.class);
+        PsiExpression mockQualifier = mock(PsiExpression.class);
+        PsiNewExpression nextMockQualifier = mock(PsiNewExpression.class);
+        PsiType mockType = mock(PsiType.class);
+        PsiMethodCallExpression mockQualifierMethodCall = mock(PsiMethodCallExpression.class);
+
+
+        // Act
+        when(mockExpression.getMethodExpression()).thenReturn(mockMethodExpression);
+        when(mockMethodExpression.getReferenceName()).thenReturn(METHOD_TO_CHECK);
```

Comment:
we should be mocking the class/java example and not directly returning the method_to_check

Reply:
Like testdata should be equal to L31

Reply:
checked in with you that this is fine as is -- you have to define the behaviour of the mock with the stub because its the logic being tested

Reply:
What the test should be testing is that it can parse through a java files with methods and classes and is still able to detect the "METHOD_TO_CHECK".

That would be the correct way of testing the logic written in the rule.
@Njeriiii could you add a follow up issue to track this?